src/combinators/mod.rs

```diff
@@ -22,6 +22,12 @@ pub trait Skip: Write {
     fn skip(self, s: usize) -> Result<Self, GenError> where Self: Sized;
 }
 
+/// Trait for `Write` types that allow skipping and reserving a slice, then writing some data,
+/// then write something in the slice we reserved using the return for our data write.
```

Comment:
Can you also add a combinator for this, with documentation that includes a small example? :)

Reply:
Yep, was planning on doing that, I preferred to push the PoC before